samples/bluetooth/per_sync/src/main.c

```diff
@@ -0,0 +1,287 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <device.h>
+#include <devicetree.h>
+#include <drivers/gpio.h>
+#include <bluetooth/bluetooth.h>
+
+/* The devicetree node identifier for the "led0" alias. */
+#define LED0_NODE DT_ALIAS(led0)
+
+#if DT_NODE_HAS_STATUS(LED0_NODE, okay)
+#define HAS_LED     1
+#define LED0        DT_GPIO_LABEL(LED0_NODE, gpios)
+#define PIN         DT_GPIO_PIN(LED0_NODE, gpios)
+#define FLAGS       DT_GPIO_FLAGS(LED0_NODE, gpios)
+#define BLINK_ONOFF K_MSEC(500)
+static struct device const   *dev;
+static struct k_delayed_work blink_work;
+static bool                  led_is_on;
+#endif
+
+#define NAME_LEN    30
+
+static bool         is_per_adv;
+static struct k_sem sem_per_adv;
+static struct k_sem sem_per_sync;
+static struct k_sem sem_per_sync_lost;
+static bt_addr_le_t per_addr;
+static uint8_t      per_sid;
+
+#if defined(HAS_LED)
+static void blink_timeout(struct k_work *work)
+{
+	led_is_on = !led_is_on;
+	gpio_pin_set(dev, PIN, (int)led_is_on);
+
+	k_delayed_work_submit(&blink_work, BLINK_ONOFF);
+}
+#endif
+
+static bool data_cb(struct bt_data *data, void *user_data)
+{
+	char *name = user_data;
+	uint8_t len;
+
+	switch (data->type) {
+	case BT_DATA_NAME_SHORTENED:
+	case BT_DATA_NAME_COMPLETE:
+		len = MIN(data->data_len, NAME_LEN - 1);
+		memcpy(name, data->data, len);
+		name[len] = '\0';
+		return false;
+	default:
+		return true;
+	}
+}
+
+static const char *phy2str(uint8_t phy)
+{
+	switch (phy) {
+	case 0: return "No packets";
+	case BT_GAP_LE_PHY_1M: return "LE 1M";
+	case BT_GAP_LE_PHY_2M: return "LE 2M";
+	case BT_GAP_LE_PHY_CODED: return "LE Coded";
+	default: return "Unknown";
+	}
+}
+
+static void scan_recv(const struct bt_le_scan_recv_info *info,
+		      struct net_buf_simple *buf)
+{
+	char le_addr[BT_ADDR_LE_STR_LEN];
+	char name[NAME_LEN];
+
+	(void)memset(name, 0, sizeof(name));
+
+	bt_data_parse(buf, data_cb, name);
+
+	bt_addr_le_to_str(info->addr, le_addr, sizeof(le_addr));
+	printk("[DEVICE]: %s, AD evt type %u, Tx Pwr: %i, RSSI %i %s "
```

Comment:
This print takes two full lines, when displaying this print on my terminal with the default terminal size configuration.
That makes it harder to to see what is happing. 

For me it would have been sufficient to only print address and device name.

Reply:
This is taken from the shell application.